packages/@aws-cdk/aws-secretsmanager/lib/secret-rotation.ts

```diff
@@ -115,16 +115,18 @@ export class SecretRotationApplication {
 
   /**
    * The semantic version of the rotation application
+   *
+   * @default use '1.1.60' for partition 'aws', '1.1.37' for partition 'aws-cn'
    */
-  public readonly semanticVersion: string;
+  public readonly semanticVersion?: string;
 
   /**
    * Whether the rotation application uses the mutli user scheme
    */
   public readonly isMultiUser?: boolean;
 
-  constructor(applicationId: string, semanticVersion: string, options?: SecretRotationApplicationOptions) {
-    this.applicationId = `arn:aws:serverlessrepo:us-east-1:297356227824:applications/${applicationId}`;
+  constructor(applicationId: string, semanticVersion?: string, options?: SecretRotationApplicationOptions) {
```

Comment:
I reverted the changes of `semanticVersion` and `applicationId`.

Making the `semanticVersion` as optional string, won’t break the complication of existing usage of `SecretRotationApplication`. Indeed the value of property `semanticVersion` will be changed.

For keeping the capability to specify the version of rotation application,  I add `SecretRotationApplicationOptions.semanticVersion` to implement it. It’s a bit strange to make the deprecated property `SecretRotationApplication.semanticVersion` is no longer used. The existing usage(`new new SecretRotationApplication('SecretsManagerRDSMariaDBRotationSingleUser', '1.1.139')`) still will break.

Reply:
Ok, let's take a slightly different approach then:

We deprecate `applicationId` and replace it with `function applicationArn(partition: string = 'aws')`, which returns the ARN based on the input partition (defaulting to 'aws'). Similarly, we deprecate and replace `semanticVersion` (`semanticVersionForPartition` or `semanticVersion2` or something)? We can leave the constructor and Options as-is -- I'd be comfortable not giving customers a way to create custom values for the CN partition account, region, and version until we have some demand for it. Optionally, we could still make the `semanticVersion` parameter optional, with some documentation that it defaults to `1.1.60` if not provided.

Then in the secret rotation, we add the `applicationArn` and `semanticVersionX` into the mapping based on the partition.

What do you think about this approach?

Reply:
sounds good! will try it.

Reply:
Nick, having a deep look at your propose.

I think `function applicationArn(partition: string = 'aws')` is not feasible. Because the partition is a variable of deployment, it’s a **Token** when synthesizing the application without certain value `aws` or `aws-cn`.

Another solution I think it’s totally deprecating the class `SecretRotationApplication` and create a new V2 version for handling this case.

Reply:
Hi @zxkane,

Sorry for the delay. I was out on holiday, and then took me a while to catch back up.

You're correct that you couldn't pass in the current `AWS::Partition` into the `applicationArn` function, but what I was thinking was just calling it twice (`applicationArn('aws')` and `applicationArn('aws-cn')` and inserting that into the mapping).

I had a few minutes, so I quickly wrote up what I was thinking and pushed it to the branch. Let me know what you think -- will this work?

Also, could you help me find the link to the official docs for these account IDs and version numbers? I can find the original ('aws') accounts and versions, but searching for the aws-cn versions doesn't appear to yield any (official) results.

Reply:
@njlynch , the secret manager doc in AWS China is incorrect. will share you the ticket link of doc.

Reply:
The changes looks good for me. Will fix the build issue.